sql/api/src/main/scala/org/apache/spark/sql/catalyst/parser/parsers.scala

```diff
@@ -50,6 +50,10 @@ abstract class AbstractParser extends DataTypeParserInterface with Logging {
     astBuilder.visitSingleTableSchema(parser.singleTableSchema())
   }
 
+  override def parseRoutineParam(sqlText: String): StructType = parse(sqlText) { parser =>
```

Comment:
so the main difference between this and `parseTableSchema` is: it supports default expression, and it fails for unsupported fields.

Reply:
yeah